File: computeError.m
function good_prediction = computeError(cPoints, dPoints, r_threshold)
% Compute L-2 norm error between cornerDetector and data set points
% The input arguments are:
%    cPoints = (x,y) points from cornerDetector
%    dPoints = (x,y) points from dataset
%    r_threshold = margin radius of error per corner
% The output argument is:
%    good_prediction = boolean of whether predicted corners are good enough

distances = 999*ones(1,4);
for i=1:4
    distances(i) = sqrt((cPoints(1,i) - dPoints(1,i))^2 + (cPoints(2,i) - dPoints(2,i))^2);
end

good_prediction = ~any(distances > r_threshold);

end

File: python/computeError.py

import numpy as np
import scipy
import matcompat

# if available import pylab (from matlibplot)
try:
    import matplotlib.pylab as plt
except ImportError:
    pass

def computeError(cPoints, dPoints):

    # Local Variables: i, cPoints, j, dPoints, error
    # Function calls: computeError, sqrt
    #% Compute L-2 norm error between cornerDetector and data set points
    #% The input arguments are:
    #%    cPoints = (x,y) points from cornerDetector
    #%    dPoints = (x,y) points from dataset
    #% The output argument is:
    #%    error = L-2 norm error sqrt((x_c - x_d)^2 + (y_c - y_d)^2)
    error = 0.
    for i in np.arange(1., 5.0):
        for j in np.arange(1., 3.0):
            error = error+(cPoints[int(i)-1,int(j)-1]-dPoints[int(i)-1,int(j)-1])**2.
            
        
    error = np.sqrt(error)
    return [error]